modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/manager/ExtensionRegistry.java

```diff
@@ -128,7 +128,7 @@ Collection<ConfigurationProvider> getConfigurationProviders(ExtensionModel exten
    * @return a {@link ConfigurationProvider}
    */
   Optional<ConfigurationProvider> getConfigurationProvider(String key) {
-    return registry.lookupByName(key);
+    return registry.lookupByName(key).filter(cp -> cp instanceof ConfigurationProvider).map(cp -> (ConfigurationProvider) cp);
```

Comment:
I added this so we get an empty optional if the object corresponding to the key is not a ConfigurationProvider.
Otherwise we would be getting class cast exceptions.
Do you think this could cause any issue?
I could also change it to avoid the lambdas if you think performance could be an issue here. But I wasn't sure if it would be premature optimization.

Reply:
If the key doesn't correspond to a COnfigurationProvider or doesn't exist then an error should be thrown. Each scenario should have a different error, with descriptive messages

Reply:
Yes, I have the check [here](https://github.com/mulesoft/mule/pull/11413/files#diff-99911b51df158c9f4dfff10192fc7e131e802732880b9a1e45cab45cf9f5db68R79). And the test [here](https://github.com/mulesoft/mule/pull/11413/files#diff-4495b18327d12c8ca6d408332ac61d600826fc0bb550b2c645acdf7d11c3a6c4R67).
But if I didn't change this line in particular, I would have had to catch class cast exceptions instead of just checking for presence.